test/integration/auditingIntegrationTests.coffee

```diff
@@ -28,3 +40,65 @@ describe "Auditing Integration Tests", ->
 
         # async test :(
         setTimeout checkAudits, 1000
+
+  describe "TLS Audit Server", ->
+
+    it "should send TLS audit messages and save (valid)", (done) ->
+
+      options =
+        cert: fs.readFileSync "test/resources/trust-tls/cert1.pem"
+        key:  fs.readFileSync "test/resources/trust-tls/key1.pem"
+        ca: [ fs.readFileSync "test/resources/server-tls/cert.pem" ]
+
+      normalSocket = new (net.Socket)
+      client = new (tls.TLSSocket)(normalSocket)
+      client.connect 5051, 'localhost', options, ->
+        messagePrependlength = testAuditMessage.length + ' ' + testAuditMessage
+        client.write messagePrependlength
+
```

Comment:
Have you tried a `client.end()` here. It may make all the difference.

Also, you aren't waiting for the data to be written to the socket here so maybe that is an issue? Perhaps try end the sockets the listen for the `end` event then only check for audits. You may be looking for audits before they have been saved.